lib/awful/permissions/init.lua

```diff
@@ -567,6 +568,101 @@ function permissions.client_geometry_requests(c, context, hints)
     end
 end
 
+--- Begins moving a client with the mouse.
+--
+-- This is the default handler for `request::mouse_move`. When a request is
+-- received, it uses `awful.mouse.resize` to initiate a mouse movement transaction
+-- that lasts so long as the specified mouse button is held.
+--
+-- @signalhandler awful.permissions.client_mouse_move
+-- @tparam client c The client
+-- @tparam string context The context
+-- @tparam table args Additional information describing the movement.
+-- @tparam number args.mouse_pos.x The x coordinate of the mouse when grabbed.
+-- @tparam number args.mouse_pos.y The y coordinate of the mouse when grabbed.
+-- @tparam number args.button The mouse button that initiated the movement.
+-- @sourcesignal client request::mouse_move
+function permissions.client_mouse_move(c, context, args)
+    if not pcommon.check(c, "client", "mouse_move", context) then return end
+
+    if not c
+        or c.fullscreen
+        or c.maximized
+        or c.type == "desktop"
+        or c.type == "splash"
+        or c.type == "dock" then
+        return
+    end
+
+    local center_pos = aplace.centered(mouse, {parent=c, pretend=true})
+    mresize(c, "mouse.move", {
+        placement = aplace.under_mouse,
+        offset = {
+            x = center_pos.x - args.mouse_pos.x,
+            y = center_pos.y - args.mouse_pos.y
+        },
+        mouse_buttons = {args.button}
+    })
+end
+
+--- Begins resizing a client with the mouse.
+--
+-- This is the default handler for `request::mouse_resize`. When a request is
+-- received, it uses `awful.mouse.resize` to initiate a mouse resizing transaction
+-- that lasts so long as the specified mouse button is held.
+--
+-- @signalhandler awful.permissions.client_mouse_resize
+-- @tparam client c The client
+-- @tparam string context The context
+-- @tparam table args Additional information describing the resizing.
+-- @tparam number args.mouse_pos.x The x coordinate of the mouse when grabbed.
+-- @tparam number args.mouse_pos.y The y coordinate of the mouse when grabbed.
+-- @tparam number args.button The mouse button that initiated the resizing.
+-- @tparam string args.corner The corner/side of the window being resized.
+-- @sourcesignal client request::mouse_resize
+function permissions.client_mouse_resize(c, context, args)
+    if not pcommon.check(c, "client", "mouse_resize", context) then return end
+
+    if not c
+        or c.fullscreen
+        or c.maximized
+        or c.type == "desktop"
+        or c.type == "splash"
+        or c.type == "dock" then
+        return
+    end
+
+    local corner_pos = aplace[args.corner](mouse, {parent = c, pretend = true})
+    mresize(c, "mouse.resize", {
+        corner = args.corner,
+        corner_lock = true,
```

Comment:
These two new parameters aren't documented. I think they should be part of the Placement “Common arguments” list.

Also, I think only `corner` should be required. `corner_lock` seems unnecessary considering the test in `placement.resize_to_mouse` is `if args.corner and args.corner_lock then`.

Reply:
Yes, you're right. It makes more sense to have this specific parameter in the function documentation rather than the "common parameters" section.